lib/pengine/rules.c

```diff
@@ -1160,44 +1182,40 @@ pe__eval_op_expr(const xmlNode *expr, const pe_rule_eval_data_t *rule_data)
 gboolean
 pe__eval_role_expr(const xmlNode *expr, const pe_rule_eval_data_t *rule_data)
 {
-    gboolean accept = FALSE;
     const char *op = NULL;
-    const char *value = NULL;
 
-    if (rule_data->role == pcmk_role_unknown) {
-        return accept;
+    if (rule_data->role == pcmk_role_unknown) { // Skip role checks for now
+        return FALSE;
     }
 
-    value = crm_element_value(expr, XML_EXPR_ATTR_VALUE);
     op = crm_element_value(expr, XML_EXPR_ATTR_OPERATION);
 
+    // @TODO Why are only Unpromoted/Promoted considered defined?
```

Comment:
Umm ... that appears to be the case. We don't have any regression tests using `#role` :-/ I'm not really sure what use it has, since location constraints accept a role attribute (which, oddly, is undocumented). Another to-do ...

Reply:
A node attribute would be a suboptimal design anyway, since a unique promotable clone could have multiple instances on a node with different roles. It looks like `#role` was never implemented. I'll just drop it.

Reply:
I'll save that for a later PR though